File: Dart/Encapsulamento/Escola/professor.dart
class Professor {
  // Atributos da classe Professor
  String _nome = "";
  String _cpf = "";
  String _idProfessor = "";

  // Normalmente, só se usa chaves quando o setter/getter tem mais de uma linha de código.
  // Para uma linha só, pode-se usar arrow function.

  // Construtor
  Professor(this._nome);

  // Getter e setter de CPF
  void set setCpf(String cpfEspecifico) => _cpf = cpfEspecifico;

  String get getCPF => _cpf; 

  // Getter e setter de Id
  void set setID(String idEspecifico) => _idProfessor = idEspecifico;

  String get getID => _idProfessor;

  // Método que retorna as informações
  String retornarInformacoesProf(){ 
    return "\nProfessor \nNome: $_nome \nCPF: $_cpf  \nID do professor: $_idProfessor\n";
  }
}

File: linguagem_java/encapsulamento/escola/Professor.java
package linguagem_java.encapsulamento.escola;

public class Professor {
    // Atributos da classe Professor
    private String nome;
    private String cpf;
    private String id;

    // Construtor
    Professor(String nomeEspecifico) {
        setNome(nomeEspecifico);
    }

    // Getter e setter de nome
    public void setNome(String nomeEspecifico) {
        this.nome = nomeEspecifico;
    }

    public String getnome() {
        return this.nome;
    }

    // Getter e setter de CPF
    public void setCpf(String cpfEspecifico) {
        this.cpf = cpfEspecifico;
    }

    public String getCpf(){ 
        return this.cpf;
    }

    // Getter e setter de ID
    public void setId(String idEspecifico) {
        this.id = idEspecifico;
    }

    public String getId(){ 
        return this.id;
    }


}
